core/src/main/java/org/jdbi/v3/BuiltInArgumentFactory.java

```diff
@@ -90,6 +91,7 @@
         register(map, Time.class, Types.TIME, PreparedStatement::setTime);
         register(map, Timestamp.class, Types.TIMESTAMP, PreparedStatement::setTimestamp);
         register(map, URL.class, Types.DATALINK, PreparedStatement::setURL);
+        register(map, URI.class, Types.VARCHAR, stringifyValue(PreparedStatement::setString));
```

Comment:
`URL` is this funky `DATALINK` type -- for symmetry, we could consider that, but I have no idea what it does or why it's used and Postgres / H2 are both perfectly happy with Strings.